### PR TITLE
add BaseReplicatedEntity and common Entity interface

### PR DESCRIPTION
**Pull request by JoshMcCullough**
In order to treat our "replicated" entities as normal entities, but still allow the id, version, etc. to carry over from the event ...

I've added a new `BaseReplicatedEntity` which should be used e.g. in Page for `Project` and `Profile` entities (entities which are not owned by Page itself). This way all entities can share `BaseRepo`.

### CODE REVIEW

File: src/main/java/com/teslagov/spring/entity/BaseReplicatedEntity.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+public abstract class BaseReplicatedEntity<I extends Serializable> implements Entity<I>
+{
+    @Id
+    @Column
+    private I id;
+
+    @Column
+    private Long version;
+
+    @Column(nullable = false, columnDefinition = "TIMESTAMP default CURRENT_TIMESTAMP")
```
Comment (fitzyjoe) [inline src/main/java/com/teslagov/spring/entity/BaseReplicatedEntity.java]: no default values for these... create should be not nullable.

File: src/main/java/com/teslagov/spring/entity/BaseReplicatedEntity.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private Instant createdDateTime;
+
+    @Column(nullable = false, columnDefinition = "TIMESTAMP default CURRENT_TIMESTAMP on update CURRENT_TIMESTAMP")
```
Comment (fitzyjoe) [inline src/main/java/com/teslagov/spring/entity/BaseReplicatedEntity.java]: no default values for these... create should be not nullable.

File: src/main/java/com/teslagov/spring/entity/BaseReplicatedEntity.java (hunk 1 above)
Comment (JoshMcCullough) [reply src/main/java/com/teslagov/spring/entity/BaseReplicatedEntity.java]: Agreed.

File: src/main/java/com/teslagov/spring/entity/BaseReplicatedEntity.java (hunk 2 above)
Comment (JoshMcCullough) [reply src/main/java/com/teslagov/spring/entity/BaseReplicatedEntity.java]: Agreed.

**Pull request merged**